[incubator/druid] overlord is broken when gCloudStorage.enabled is true

<!-- Thanks for filing an issue! Before hitting the button, please answer these questions. It's helpful to search the existing GitHub issues first. It's likely that another user has already reported the issue you're facing, or it's a known issue that we're already aware of 

Fill in as much of the template below as you can.  If you leave out information, we can't help you as well.

Be ready for followup questions, and please respond in a timely manner. If we can't reproduce a bug or think a feature already exists, we might close your issue.  If we're wrong, PLEASE feel free to reopen it and explain why.
-->

**Describe the bug**
Overlord is failing to start if gCloudStorage.enabled is true. 
```
2020-08-07T10:09:23,404 ERROR [main] org.apache.druid.cli.CliOverlord - Error when starting up.  Failing.
com.google.inject.ProvisionException: Unable to provision, see the following errors:
1) Error in custom provider, org.apache.druid.java.util.common.RE: server initialization exception
  at org.apache.druid.server.initialization.jetty.JettyServerModule.getServer(JettyServerModule.java:165) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModul
e -> org.apache.druid.server.initialization.jetty.JettyServerModule)
  at org.apache.druid.server.initialization.jetty.JettyServerModule.getServer(JettyServerModule.java:165) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModul
e -> org.apache.druid.server.initialization.jetty.JettyServerModule)
  while locating org.eclipse.jetty.server.Server
1 error
        at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1028) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1050) ~[guice-4.1.0.jar:?]
        at org.apache.druid.guice.LifecycleModule$2.start(LifecycleModule.java:141) ~[druid-core-0.19.0.jar:0.19.0]
        at org.apache.druid.cli.GuiceRunnable.initLifecycle(GuiceRunnable.java:115) [druid-services-0.19.0.jar:0.19.0]
        at org.apache.druid.cli.ServerRunnable.run(ServerRunnable.java:63) [druid-services-0.19.0.jar:0.19.0]
        at org.apache.druid.cli.Main.main(Main.java:113) [druid-services-0.19.0.jar:0.19.0]
Caused by: org.apache.druid.java.util.common.RE: server initialization exception
        at org.apache.druid.server.initialization.jetty.JettyServerModule.makeAndInitializeServer(JettyServerModule.java:404) ~[druid-server-0.19.0.jar:0.19.0]
        at org.apache.druid.server.initialization.jetty.JettyServerModule.getServer(JettyServerModule.java:165) ~[druid-server-0.19.0.jar:0.19.0]
        at org.apache.druid.server.initialization.jetty.JettyServerModule$$FastClassByGuice$$ff9924d8.invoke(<generated>) ~[druid-server-0.19.0.jar:0.19.0]
        at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.doProvision(ProviderMethod.java:264) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.ProviderMethod$Factory.provision(ProviderMethod.java:401) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.ProviderMethod$Factory.get(ProviderMethod.java:376) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:194) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1019) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1085) ~[guice-4.1.0.jar:?]
        at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1015) ~[guice-4.1.0.jar:?]
        ... 5 more
Caused by: com.google.inject.ProvisionException: Unable to provision, see the following errors:
1) Error in custom provider, java.io.IOException: Error reading credential file from environment variable GOOGLE_APPLICATION_CREDENTIALS, value '/var/secrets/google/key.json': File does not exist.
  at org.apache.druid.common.gcp.GcpModule.getHttpRequestInitializer(GcpModule.java:60) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.common.gcp.GcpModule)
  at org.apache.druid.common.gcp.GcpModule.getHttpRequestInitializer(GcpModule.java:60) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.common.gcp.GcpModule)
  while locating com.google.api.client.http.HttpRequestInitializer
    for the 3rd parameter of org.apache.druid.storage.google.GoogleStorageDruidModule.getGoogleStorage(GoogleStorageDruidModule.java:107)
  at org.apache.druid.storage.google.GoogleStorageDruidModule.getGoogleStorage(GoogleStorageDruidModule.java:107) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.google.Goog
leStorageDruidModule)
  while locating org.apache.druid.storage.google.GoogleStorage
    for the 2nd parameter of org.apache.druid.storage.google.GoogleTaskLogs.<init>(GoogleTaskLogs.java:55)
  at org.apache.druid.storage.google.GoogleStorageDruidModule.configure(GoogleStorageDruidModule.java:96) (via modules: com.google.inject.util.Modules$OverrideModule -> org.apache.druid.storage.google.GoogleStorag
eDruidModule)
```

Missing credential is the problem so we should add configMap to overlord deployment if gCloudStorage.enabled=true

--- a/incubator/druid/Chart.yaml
+++ b/incubator/druid/Chart.yaml
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 0.19.0
 description: Apache Druid is a high performance real-time analytics database.
 name: druid
-version: 0.2.10
+version: 0.2.11
 home: https://druid.apache.org/
 icon: https://druid.apache.org/img/favicon.png
 maintainers:

--- a/incubator/druid/templates/overlord/deployment.yaml
+++ b/incubator/druid/templates/overlord/deployment.yaml
@@ -56,6 +56,17 @@ spec:
               port: {{ .Values.overlord.port }}
           resources:
 {{ toYaml .Values.overlord.resources | indent 12 }}
+          volumeMounts:
+  {{- if .Values.gCloudStorage.enabled }}
+            - name: google-cloud-key
+              mountPath: /var/secrets/google
+  {{- end }}
+      volumes:
+  {{- if .Values.gCloudStorage.enabled }}
+      - name: google-cloud-key
+        secret:
+          secretName: {{ .Values.gCloudStorage.secretName }}
+  {{- end }}
     {{- with .Values.overlord.nodeSelector }}
       nodeSelector:
 {{ toYaml . | indent 8 }}

